### PR TITLE
static check fixes

### PR DESCRIPTION
**Pull request by coryschwartz**
various fixes for staticcheck

For context: https://github.com/protocol/.github/issues/41

### CODE REVIEW

File: p2p/host/relay/autorelay.go [hunk 1]
```diff
@@ -247,7 +247,15 @@ func (ar *AutoRelay) discoverRelays(ctx context.Context) ([]peer.AddrInfo, error
 
 	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
-	return discovery.FindPeers(ctx, ar.discover, RelayRendezvous, discovery.Limit(1000))
+	var ret []peer.AddrInfo
+	ch, err := ar.discover.FindPeers(ctx, RelayRendezvous, discovery.Limit(1000))
```
Comment (coryschwartz) [inline p2p/host/relay/autorelay.go:251]: This should be equivilent to the FindPeers util function in go-libp2p-discovery, to convert the channel into a slice.

File: p2p/host/relay/relay.go [hunk 2]
```diff
@@ -20,7 +20,31 @@ func Advertise(ctx context.Context, advertise discovery.Advertiser) {
 	go func() {
 		select {
 		case <-time.After(AdvertiseBootDelay):
-			discovery.Advertise(ctx, advertise, RelayRendezvous, discovery.TTL(AdvertiseTTL))
+			go func() {
```
Comment (coryschwartz) [inline p2p/host/relay/relay.go:23]: This is brought in from go-libp2p-discovery.

If it's better to use the Util Functions from there I can swtich back to using them.

File: p2p/protocol/identify/id.go [hunk 3]
```diff
@@ -368,7 +368,7 @@ func (ids *IDService) identifyConn(c network.Conn, signal chan struct{}) {
 
 	// ok give the response to our handler.
 	if err = msmux.SelectProtoOrFail(ID, s); err != nil {
-		log.Event(context.TODO(), "IdentifyOpenFailed", c.RemotePeer(), logging.Metadata{"error": err})
+		// log.Event(context.TODO(), "IdentifyOpenFailed", c.RemotePeer(), logging.Metadata{"error": err})
```
Comment (coryschwartz) [inline p2p/protocol/identify/id.go:371]: Is there a metric that should be emitted here in leiu of depreciated go-log Events?

File: p2p/host/routed/routed.go [hunk 4]
```diff
@@ -131,7 +130,7 @@ func logRoutingErrDifferentPeers(ctx context.Context, wanted, got peer.ID, err e
 	lm["error"] = err
 	lm["wantedPeer"] = func() interface{} { return wanted.Pretty() }
 	lm["gotPeer"] = func() interface{} { return got.Pretty() }
-	log.Event(ctx, "routingError", lm)
+	// log.Event(ctx, "routingError", lm)
```
Comment (marten-seemann) [inline p2p/host/routed/routed.go:133]: Why is this code commented?

File: p2p/net/mock/mock_test.go [hunk 5]
```diff
@@ -15,18 +15,19 @@ import (
...
 )
 
-func randPeer(t *testing.T) peer.ID {
-	p, err := test.RandPeerID()
-	if err != nil {
-		t.Fatal(err)
-	}
-	return p
-}
+// Unused code:
```
Comment (marten-seemann) [inline p2p/net/mock/mock_test.go:22]: Can we delete this?

File: p2p/host/relay/relay.go (hunk 2 above)
Comment (Stebalien) [reply p2p/host/relay/relay.go:23]: Well, those functions exist so we don't need to duplicate this code. What was the motivation for this change?

File: p2p/host/routed/routed.go (hunk 4 above)
Comment (Stebalien) [reply p2p/host/routed/routed.go:133]: This entire function should be deleted and replaced with a call to `log.Errorw`.

Side note: please (almost) never comment out code unless you're actively debugging.

**Review by Stebalien — CHANGES_REQUESTED** [review]

File: p2p/protocol/identify/id.go (hunk 3 above)
Comment (Stebalien) [reply p2p/protocol/identify/id.go:371]: Just a structured log message.

File: p2p/host/relay/relay.go (hunk 2 above)
Comment (Stebalien) [reply p2p/host/relay/relay.go]: Ah... I see.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**